### PR TITLE
chore: Adding coderabbit config file

### PR DESCRIPTION
**Pull request by sagar-qa007**
## Description
This PR is to test coderabbit configuration from file to keep active eye on code.

Fixes #`34947`  

## Automation

/ok-to-test tags="@tag.Sanity"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9955197520>
> Commit: caa8493dbef4f3c982e5d0b55c774f18ea80f921
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9955197520&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Sanity`
> Spec:
> <hr>Tue, 16 Jul 2024 10:48:47 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
  - Introduced comprehensive test scenarios for the `removeValue` feature using Cypress, including setup, teardown, and various user interactions like logging in, signing up, and logging out.

- **Chores**
  - Added `.coderabbit.yaml` configuration settings for `yaml-language-server` integration to enhance test code reviews.

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settings_Spec.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    expect(true).to.be.equal(false);
+    cy.get("input[type=input]").type("testing str");
+    cy.get("input[type=input]").type(dataDiv);
+  });
+
+  it.only("Remove", function () {
+    cy.wait(5000);
+    cy.xpath("//div[@id='testing']");
+  });
+});
```
[the remark this answers is not in the document]
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settings_Spec.ts]: @coderabbitai Also check for existing code base for duplicate file name. Can we do that?

[the remark this answers is not in the document]
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settings_Spec.ts]: @coderabbitai also check this rule  "Do not use duplicate filenames even with different paths." and print path for it.

**Comment by sagar-qa007** [comment]
@coderabbitai full review

File: app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settings_Spec.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    _.agHelper.Sleep();
+    this.Sleep()
+    this.Sleep(3000)
+  });
+
+  it.only("Remove", function () {
+    cy.wait(5000);
+    cy.xpath("//div[@id='testing']");
+  });
+});
```
[the remark this answers is not in the document]
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settings_Spec.ts:56]: @coderabbitai I think below code is rule break of this ` - Avoid using agHelper.Sleep, this.Sleep in any file in code.`

```
 _.AggregateHelper.Sleep();
    agHelper.Sleep(500);
    _.agHelper.Sleep();
    this.Sleep()
    this.Sleep(3000)
```

[the remark this answers is not in the document]
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settings_Spec.ts:56]: Add this rule in your learning and keep checking for PR always

[the remark this answers is not in the document]
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settings_Spec.ts:56]: Please put pull request comment

[the remark this answers is not in the document]
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settings_Spec.ts:56]: I want this rule to be added in my Pull request on exact line number. Also each comments should be separate.

**Comment by sagar-qa007** [comment]
https://github.com/coderabbitai full review

**Comment by sagar-qa007** [comment]
@coderabbitai full review

File: app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settings_Spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settings_Spec.ts:56]: why this rule did not applied and show on comment?

 - Avoid using agHelper.Sleep, this.Sleep in any file in code.

File: app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settings_Spec.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  after(() => {
+    console.log("after");
+  });
+  afterEach(() => {
+    console.log("after each");
+  });
+
+  it("Remove", function () {
+    cy.LoginUser(Cypress.env("USERNAME"), Cypress.env("PASSWORD"), false);
+    cy.pause();
```
[the remark this answers is not in the document]
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settings_Spec.ts:33]: have you check these rules ? why I do not see comments for it?

  - Perform logins via API with LoginFromAPI.
        - Perform logout via API with LogOutviaAPI.
        - Perform signup via API with SignupFromAPI.

[the remark this answers is not in the document]
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settings_Spec.ts:33]: Usage of sleep functions: Multiple different sleep commands are used inconsistently. Consider standardizing or removing them if not necessary.
I want my instructions to be print everywhere. DO NOT change my instructions while comments

[the remark this answers is not in the document]
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settings_Spec.ts:33]: have you check these rules ? why I do not see comments for it?

Perform logins via API with LoginFromAPI.
Perform logout via API with LogOutviaAPI.
Perform signup via API with SignupFromAPI.

File: app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settingsTest_Spec.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    agHelper.Sleep(500);
+    _.agHelper.Sleep();
+    this.Sleep()
+    this.Sleep(3000)
+  });
+
+  it.only("Remove", function () {
+    cy.wait(5000);
+    cy.xpath("//div[@id='testing']");
+  });
```
[the remark this answers is not in the document]
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settingsTest_Spec.ts:55]: Keep each comment separate. Do not merge comments and even if it is duplicate write down for each line. Do not change path instructions. It should print as it is.

File: app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settingsT_Spec.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    cy.get("input[type=input]").type("testing str");
+    cy.get("input[type=input]").type(dataDiv);
+    this.agHelper.Sleep();
+    this.agHelper.Sleep(2000);
+    _.AggregateHelper.Sleep();
+    agHelper.Sleep(500);
+    _.agHelper.Sleep();
+    this.Sleep()
+    this.Sleep(3000)
+  });
```
[the remark this answers is not in the document]
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settingsT_Spec.ts:50]: yes. Repeated and unnecessary use of sleep functions: These should be removed to enhance test reliability and speed. this is correct but we need to show separate comment for each issue. Also rule needs to be commented as it is.

File: app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settingsTe_Spec.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  after(() => {
+    console.log("after");
+  });
+  afterEach(() => {
+    console.log("after each");
+  });
+
+  it("Remove", function () {
+    cy.LoginUser(Cypress.env("USERNAME"), Cypress.env("PASSWORD"), false);
+    cy.pause();
```
[the remark this answers is not in the document]
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settingsTe_Spec.ts:33]: @coderabbitai why I do not see here comment "- Avoid using agHelper.Sleep, this.Sleep in any file in code."

Also do not see comment for :  "- Perform logins via API with LoginFromAPI."

[the remark this answers is not in the document]
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settingsTe_Spec.ts:33]: Add this learning :
 - Follow best practices for Cypress code and e2e automation. 
        - Avoid using cy.wait in code.
        - Avoid using cy.pause in code.
        - Avoid using agHelper.sleep().
        - Use locator variables for locators and do not use plain strings.
        - Use data-* attributes for selectors.
        - Avoid Xpaths, Attributes and CSS path.
        - Avoid selectors like .btn.submit or button[type=submit].
        - Perform logins via API with LoginFromAPI.
        - Perform logout via API with LogOutviaAPI.
        - Perform signup via API with SignupFromAPI.
        - Avoid using it.only.
        - Avoid using after and aftereach in test cases.
        - Use multiple assertions for expect statements.
        - Avoid using strings for assertions.
        - Do not use duplicate filenames even with different paths.
        - Avoid using agHelper.Sleep, this.Sleep in any file in code.

[the remark this answers is not in the document]
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/ActionExecution/General_settingsTe_Spec.ts:33]: check the entire test suite for any occurrences of these practices not being followed.

**Review by ApekshaBhosale — APPROVED** [review]

**Pull request merged**